### ISSUE TITLE
bug: Error occurs when dynamically changing picker column options

### ISSUE THREAD

**Issue by ryaa** [ionic-team/ionic-framework#21763]
# Bug Report

**Ionic version:**

[ ] **4.x**
[x] **5.x**

**Current behavior:**

An error occurs when dynamically changing picker column options and the picker UI is not updated as required.

**Expected behavior:**

picker UI must be updated accordingly and no errors must occur.

**Steps to reproduce:**

1) open the link to stackblitz (see below)
2) click/tap OPEN PICKER button
3) change the value in the first column
Note that there is an error shown in console "Error: Cannot read property 'duration' of undefined" and values in the second picker column are not updated.

**Related code:**
See https://stackblitz.com/edit/ionic-angular-v5-y1p2sa?file=src%2Fapp%2Fapp.component.ts

**Other information:**
The problem is that the picker component does not correctly handle the use case when the number of options in a column changed. It loops through the options that were before the update and failed because the number of options has been changed/decreased - see below

<img width="492" alt="Screenshot 2020-07-16 at 11 33 15" src="https://user-images.githubusercontent.com/3608222/87651498-a0af0f00-c75b-11ea-953d-94f6a8b915f2.png">

<img width="1038" alt="Screenshot 2020-07-16 at 11 33 45" src="https://user-images.githubusercontent.com/3608222/87651526-ac9ad100-c75b-11ea-9330-683145acbee5.png">

**Ionic info:**
```
Ionic:

   Ionic CLI                     : 6.9.2 (/Users/alexryltsov/.nvm/versions/node/v12.16.3/lib/node_modules/@ionic/cli)
   Ionic Framework               : @ionic/angular 5.1.1
   @angular-devkit/build-angular : 0.901.7
   @angular-devkit/schematics    : 9.1.7
   @angular/cli                  : 9.1.7
   @ionic/angular-toolkit        : 2.2.0

Capacitor:

   Capacitor CLI   : 2.1.1
   @capacitor/core : 2.1.1

Utility:

   cordova-res : not installed
   native-run  : not installed

System:

   NodeJS : v12.16.3 (/Users/alexryltsov/.nvm/versions/node/v12.16.3/bin/node)
   npm    : 6.14.5
   OS     : macOS Catalina
```

**Comment by snakone** [comment]
having the same issue here, any solution so far?

**Comment by obnijnil** [comment]
This comment from the related issue may be a solution for NOW. https://github.com/ionic-team/ionic-framework/issues/17664#issuecomment-531177999

**Comment by thetaPC** [comment]
Thank you for submitting the issue! The fix has been merged through PR ionic-team/ionic-framework#27359.
